tests_e2e/tests/agent_update/rsm_update.py

```diff
@@ -0,0 +1,143 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# BVT for the agent update scenario
+#
+import json
+import os.path
+import time
+
+import requests
+from assertpy import assert_that
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.compute.models import VirtualMachine
+
+from tests_e2e.tests.lib.agent_test import AgentTest
+from tests_e2e.tests.lib.agent_test_context import AgentTestContext
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine import VmMachine
+
+
+class RsmUpdateBvt(AgentTest):
+
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = None
+
+    def run(self):
+        self._ssh_client = SshClient(
+            ip_address=self._context.vm_ip_address,
+            username=self._context.username,
+            private_key_file=self._context.private_key_file)
+
+        # Allow agent to send supported feature flag
+        time.sleep(60)
+
+        log.info("*******Verifying the Agent Downgrade scenario*******")
+        self._prepare_rsm_update("1.3.0.0")
+        self._prepare_agent()
+
+        # Verify downgrade scenario
+        self._verify_guest_agent_update("1.3.0.0")
+
+        # Verify upgrade scenario
+        log.info("*******Verifying the Agent Upgrade scenario*******")
+        self._prepare_rsm_update("1.3.1.0")
+        self._verify_guest_agent_update("1.3.1.0")
+
+        # verify no version update. There is bug in CRP and will enable once it's fixed
+        # log.info("*******Verifying the no version update scenario*******")
+        # self._prepare_rsm_update("1.3.1.0")
+        # self._verify_guest_agent_update("1.3.1.0")
+
+    def _prepare_agent(self):
+        local_path = self._context.test_source_directory/"tests"/"scripts"/"agent-python"
+        remote_path = self._context.remote_working_directory/"agent-python"
+        self._ssh_client.copy(local_path, remote_path)
+        python = self._ssh_client.run_command(f"sudo {remote_path}").rstrip()
+        version_file_dir = self._ssh_client.run_command(f"{python} -c 'import azurelinuxagent.common.version as v; import os; print(os.path.dirname(v.__file__))'").rstrip()
+        version_file__full_path = os.path.join(version_file_dir, "version.py")
+        # E0401: W1401: Anomalous backslash in string: '\s'. String constant might be missing an r prefix. (anomalous-backslash-in-string).
+        self._ssh_client.run_command(f"sudo sed -E -i \"s/AGENT_VERSION\s+=\s+'[0-9.]+'/AGENT_VERSION = '1.0.0.0'/\" {version_file__full_path}")  # pylint: disable=W1401
```

Comment:
instead of executing command by command, could you join all those commands in a single script and do one single ssh call? (will reduce the change of connectivity issues)